rclcpp_components/CMakeLists.txt

```diff
@@ -20,6 +20,15 @@ find_package(composition_interfaces REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(rcpputils REQUIRED)
 
+# Add an interface library that can be dependend upon by libraries who register components
+add_library(component INTERFACE)
```

Comment:
Also, should this target name be something like `component_interface` or something? We have "libcomponent_manager" and "libcomponent", but this is an "interface library". Maybe it's redundant, but I just wanted to bring it up.

Reply:
> Also, should this target name be something like component_interface

I would recommend against adding `interface` because downstream users don't need to know that to use it. For example I'm pretty sure `Eigen3::Eigen` is an interface library instead of a shared one, but regardless I'm going to pass it into a call to `target_link_libraries`. Leaving off the `interface` suffix leaves the door open to making this a shared library in the future too.